hermes/src/proof_store.rs

```diff
@@ -0,0 +1,207 @@
+use {
+    crate::db::{
+        Db,
+        DbRecord,
+        RequestTime,
+        UnixTimestamp,
+    },
+    anyhow::{
+        anyhow,
+        Result,
+    },
+    borsh::{
+        BorshDeserialize,
+        BorshSerialize,
+    },
+    pyth_sdk::{
+        Price,
+        PriceFeed,
+        PriceIdentifier,
+    },
+    pyth_wormhole_attester_sdk::{
+        BatchPriceAttestation,
+        PriceAttestation,
+        PriceStatus,
+    },
+    serde::{
+        Deserialize,
+        Serialize,
+    },
+    std::{
+        collections::{
+            HashMap,
+            HashSet,
+        },
+        ops::{
+            Deref,
+            DerefMut,
+        },
+    },
+    wormhole::VAA,
+};
+
+
+pub enum ProofUpdate {
+    Vaa(Vec<u8>),
+}
+
+pub enum ProofType {
+    BatchVaa,
+}
+
+// TODO: Is this the right type for the proof?
+#[derive(
+    Clone,
+    Default,
+    PartialEq,
+    PartialOrd,
+    Debug,
+    Serialize,
+    Deserialize,
+    BorshSerialize,
+    BorshDeserialize,
+)]
+pub struct Proof(pub Vec<Vec<u8>>);
+
+impl Deref for Proof {
+    type Target = Vec<Vec<u8>>;
+
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+impl DerefMut for Proof {
+    fn deref_mut(&mut self) -> &mut Self::Target {
+        &mut self.0
+    }
+}
+
+
+// TODO: We need to add more metadata to this struct.
+#[derive(Clone, Default, PartialEq, Debug, BorshSerialize, BorshDeserialize)]
+pub struct PriceInfo {
+    pub price_feed:   PriceFeed,
+    pub vaa_bytes:    Vec<u8>,
+    pub publish_time: UnixTimestamp,
+}
+
+#[derive(Clone, Default, Serialize, Deserialize)]
+pub struct PriceFeedsWithProof {
+    pub price_feeds: HashMap<PriceIdentifier, PriceFeed>,
+    pub proof:       Proof,
+}
```

Comment:
this struct and Proof are a little confusing to me for two reasons:
1. It looks like the `Proof` contains the update data itself in addition to the proof itself, which makes the name "proof" a little confusing. It's really the entire verifiable update. Probably also worth documenting that the price_feeds are extracted from the proof and don't need to be recombined with the proof later.
2. Proof containing a `Vec<Vec<u8>>` seems like an odd mix between generic and specific. `Vec<u8>` would be the most generic form imo. However, I think you have `Vec<Vec<u8>>` because the contracts take an array of byte arrays? Maybe worth documenting why this is the format in that case.

Reply:
About 1. You are right. Do you have a better name for it? I'll use UpdateData to be more clear.
About 2. You are right and David also mentioned it. The contracts will take it and also with the current batch vaa approach the proof is an array of byte arrays.

Reply:
per David's suggestion in another thread, I changed the proof type to have all proofs and now vec<vec<u8>> is specific to batch_vaa.